cmd/ocitools/generate.go

```diff
@@ -305,61 +311,81 @@ func setupSpec(g *generate.Generator, context *cli.Context) error {
 		}
 	}
 
-	var sd string
-	var sa, ss []string
+	err := addSeccomp(*g, context)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addSeccomp(g generate.Generator, context *cli.Context) error {
+	seccompDefault := context.String("seccomp-default")
+	seccompArch := context.String("seccomp-arch")
+	seccompKill := context.String("seccomp-kill")
+	seccompTrace := context.String("seccomp-trace")
+	seccompErrno := context.String("seccomp-errno")
+	seccompTrap := context.String("seccomp-trap")
+	seccompAllow := context.String("seccomp-allow")
 
-	if context.IsSet("seccomp-default") {
-		sd = context.String("seccomp-default")
+	// Set the DefaultAction of seccomp
+	if seccompDefault == "" {
+		seccompDefault = "errno"
 	}
 
-	if context.IsSet("seccomp-arch") {
-		sa = context.StringSlice("seccomp-arch")
+	err := g.SetDefaultSeccompAction(seccompDefault)
+	if err != nil {
+		return err
 	}
 
-	if context.IsSet("seccomp-syscalls") {
-		ss = context.StringSlice("seccomp-syscalls")
+	// Add the additional architectures permitted to be used for system calls
+	if seccompArch == "" {
+		seccompArch = "amd64,x86,x32" // Default Architectures
 	}
 
-	if sd == "" && len(sa) == 0 && len(ss) == 0 {
-		return nil
+	architectureArgs := strings.Split(seccompArch, ",")
+	err = g.SetSeccompArchitectures(architectureArgs)
+	if err != nil {
+		return err
 	}
 
-	// Set the DefaultAction of seccomp
-	if context.IsSet("seccomp-default") {
-		if err := g.SetLinuxSeccompDefault(sd); err != nil {
+	if seccompKill != "" {
```

Comment:
Can we use `IsSet` instead of comparing with the empty string here?  It would be nice to die with an error (instead of silently ignoring) folks that call `generate --seccomp-kill ''`.  Or maybe that should clear the kill action?  But I'm pretty sure silently ignoring is not what we want.